Sources/Basics/Triple+Basics.swift

```diff
@@ -147,7 +147,7 @@ extension Triple {
         case .wasi:
             return ".wasm"
         default:
-            fatalError("Cannot create dynamic libraries for os \"\(os)\".")
```

Comment:
Can we emit a diagnostic message as a warning here instead?

Reply:
Sounds resonable to me

Reply:
If it's an unknown OS, an actionable fix is to submit a feature request on SwiftPM to add support for such OS. Otherwise it seems wrong that we're silently using a file extension that can be incorrect without giving users any clues as to what could be wrong.

Reply:
If you don't want to see this warning for `armv7em-none-none-eabi`, I think we should distinguish here `none` and triples that we just don't know how to handle.

Reply:
> If it's an unknown OS, an actionable fix is to submit a feature request on SwiftPM to add support for such OS.

Ideally, that's right, but it does not immediately suppress the warning and they need to wait for the next release including the fix.

If we go emitting diagnostics like this, I think we should have an escape hatch for their immediate fix and emit messages recommending them to add support for such OS. Otherwise, such warnings would be too annoying, IMO.

Reply:
@bnbarham WDYT?

Reply:
Any thoughts on what a possible escape hatch could be?

Reply:
I was thinking of providing ways to explicitly configure their expected file extension as an escape hatch via environment variable or configuration file

Reply:
Full triple instead of platform because we expect it may change between the specified OS?

I'm not sure this is really the right place for the warning anyway though. If a platform is unsupported, then it seems likely we'd fail elsewhere. IMO we should either decide that we should have a warning for that case in general (outside of this particular call) with the escape hatch for known-customizable cases, or not diagnose.

Reply:
> Full triple instead of platform because we expect it may change between the specified OS?

Exactly. e.g. only `none` OS information does not help decide the extension.

> IMO we should either decide that we should have a warning for that case in general (outside of this particular call) with the escape hatch for known-customizable cases, or not diagnose.

Yeah, totally agree. Adding such escape hatch sounds like the right way to go because such flexibility would be quite beneficial for Embedded scenarios. (BTW Rust toolchain allows developers to have [flexible target specification](https://github.com/rust-lang/rfcs/blob/master/text/0131-target-specification.md) and we can learn from them)

Reply:
Fair enough, then this can go in as is unless Ben has any objections, and it needs a test.

Reply:
The isDarwin check should be changed to isApple. `<arch>-apple-<os>` should be almost always be a dylib